tests/unit/storage/utils.js

```diff
@@ -0,0 +1,48 @@
+const assert = require('assert');
+const { serializeCircJSON } = require('../../../lib/storage/utils');
+
+const jsonObj = {
+    key1: 'foo',
+    key2: 1,
+    key3: { key4: 5 },
+};
+
+describe.only('SerializeCircJSON', () => {
+    it('should stringify circular JSON', () => {
+        try {
+            const circularJSON = Object.assign(jsonObj, {});
+            circularJSON.key5 = circularJSON;
+            const json = JSON.stringify(circularJSON, serializeCircJSON());
+            const parsedJSON = JSON.parse(json);
+            assert.strictEqual(parsedJSON.key1, 'foo');
+            assert.strictEqual(parsedJSON.key2, 1);
+            assert.deepStrictEqual(parsedJSON.key3, { key4: 5 });
+            assert.strictEqual(parsedJSON.key5, '[Circular ~. ]');
+        } catch (e) {
+            assert.ifError(e);
+        }
+    });
+
+    it('should stringify valid JSON', () => {
+        try {
+            const validJSON = Object.assign(jsonObj, {});
+            const json = JSON.stringify(validJSON, serializeCircJSON());
+            const parsedJSON = JSON.parse(json);
+            assert.strictEqual(parsedJSON.key1, 'foo');
+            assert.strictEqual(parsedJSON.key2, 1);
+            assert.deepStrictEqual(parsedJSON.key3, { key4: 5 });
+        } catch (e) {
+            assert.ifError(e);
+        }
+    });
+
+    it('should fail to stringify invalid JSON', () => {
+        try {
+            //eslint-disable-next-line
+            const invalidJSON = { in$valid: value };
```

Comment:
Why is this object invalid or non convertible to JSON?

Reply:
It's invalid because 'value' isn't defined anywhere